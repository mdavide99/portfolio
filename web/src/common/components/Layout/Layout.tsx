import { Footer } from "../Footer";
import { Head } from "../Head";

function Layout({ children }: any) {
  return (
    <div className="flex flex-col justify-between min-h-screen bg-neutral-50 dark:bg-neutral-900">
      <Head />
      <main>{children}</main>
      <Footer />
    </div>
  );
}

export { Layout };
