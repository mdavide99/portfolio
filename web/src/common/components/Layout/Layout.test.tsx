import { Layout } from "./Layout";
import { render } from "@testing-library/react";
import { Router } from "next/router";

jest.mock("next/router", () => ({
    useRouter() {
        return {
            route: "/",
            pathname: "",
            query: "",
            asPath: "",
			events:{on: jest.fn}
        };
    },
}));

describe('Layout',()=>{
  it('renders Layout correctly', () => {
		const { container } = render(
		<Layout/>
		);
		expect(container).toMatchSnapshot();
	});
})