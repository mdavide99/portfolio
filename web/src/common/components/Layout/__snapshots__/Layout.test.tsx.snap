// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders Layout correctly 1`] = `
<div>
  <div
    class="flex flex-col justify-between min-h-screen bg-neutral-50 dark:bg-neutral-900"
  >
    <div
      class="container flex items-center justify-between mx-auto px-4 h-20 mb-10"
      id="head"
    >
      <div
        class="flex items-center text-3xl sm:text-4xl font-extrabold group"
        id="logo"
      >
        <a
          href="/"
        >
          <span
            class="group-hover:text-neutral-50 dark:group-hover:text-neutral-900 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear"
          >
            // 
          </span>
          <span
            class="group-hover:text-blue-500 dark:group-hover:text-fuchsia-500 group-hover:transition-colors duration-1000 delay-75 group-hover:duration-500 group-hover:ease-linear tracking-wide"
          >
            Davide
          </span>
        </a>
      </div>
      <button>
        <svg
          class="sm:hidden dark:text-neutral-50 text-xl hover:text-blue-500 hover:dark:text-fuchsia-500 hover:transition-color duration-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          />
        </svg>
      </button>
      <div
        class="fixed inset-x-0 bottom-0 top-0 pb-40 sm:pb-0 sm:static flex flex-col justify-center sm:flex-row items-center bg-neutral-50 dark:bg-neutral-900 sm:bg-transparent hidden sm:block "
      >
        <button>
          <svg
            class="absolute top-7 right-4 sm:hidden dark:text-neutral-50 text-xl hover:text-blue-500 hover:dark:text-fuchsia-500 hover:transition-color duration-500"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 352 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </button>
        <a
          class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
          href="/"
        >
          about
        </a>
        <a
          class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
          href="/projects"
        >
          projects
        </a>
        <a
          class="text-4xl sm:text-base my-3 sm:my-0 mx-5 text-gray-500 dark:text-neutral-100 hover:text-gray-900 hover:dark:text-white hover:transition-color hover:underline decoration-blue-500 dark:decoration-fuchsia-500 decoration-4 sm:decoration-2 underline-offset-8"
          href="/contact"
        >
          contact
        </a>
      </div>
    </div>
    <main
      class="container mx-auto px-4"
    />
    <footer
      class="container mx-auto px-4"
    >
      <div
        class="flex justify-between items-center py-4"
      >
        <div>
          <span
            class="font-light"
          >
            Copyright Â© Davide Marinotto
          </span>
        </div>
        <div
          class="flex"
        >
          <a
            class="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
            href="https://github.com/mdavide99"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
          <a
            class="last:mr-0 mr-4 cursor-pointer hover:transition-all text-xl hover:text-blue-600 dark:hover:text-fuchsia-600 text-gray-500 dark:text-neutral-50"
            href="https://www.linkedin.com/in/davide-marinotto/"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              />
            </svg>
          </a>
          <button
            class="text-lg ml-2"
          >
            <svg
              class="text-neutral-600 hover:text-blue-600"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
              />
            </svg>
          </button>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
